aten/src/ATen/cuda/CUDANumerics.cuh

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include <cuda.h>
+#include <limits.h>
+#include <assert.h>
+#include <typeinfo>
+
+// CUDANumerics.cuh contains mathematical functions that are either
+// not in the std namespace or are specialized for compilation
+// with CUDA NVCC or CUDA NVRTC. This header is derived from the
+// legacy THCNumerics.cuh.
+
+namespace at{
+
+template <typename T>
+struct numerics {
+};
+
+template <typename scalar_t>
+static inline __host__ __device__ scalar_t powi(scalar_t a, scalar_t b) {
```

Comment:
I'm not convinced that the usefulness of square and multiply merits putting it in a generic file like this. The std::pow overloads work with integers. If there's a kernel that requires square-and-multiply, we should move this there.